### DISCUSSION TITLE
How to set the proper Solid fill?

### DISCUSSION THREAD

**Discussion by StephanMeijer** [exceljs/exceljs#1658] [Q&A] [answered]
When using solid fill, the cell seems to become black entirely. It might be the case because I am using LibreOffice. Google Spreadsheets does not color the cell at all.

**Comment by Rycochet** [comment]
I recently had cause to use a solid fill myself, and no problem on it (sets an off-red background) -

```javascript
cell.fill = {
	type: "pattern",
	pattern: "solid",
	fgColor: {
		argb: "FFFF7D7D"
	},
	bgColor: {
		argb: "FF000000"
	}
}
```
Could you have done it slightly differently? (As a side note, Excel can really mess up with fills on empty cells at the end of rows - it's ugly).

**Comment by guyonroche** [comment]
@StephanMeijer I've just published 0.4.2, which I'm hoping will resolve this issue. I've fixed an issue with handling themes which can cause the effects you're seeing.
Could you check whether this has fixed your problem?

**Comment by IT-MikeS** [comment]
@guyonroche I have the same issue even after updating to 0.4.2 also I am using Excel 2016 to read the output xlsx

**Comment by IT-MikeS** [comment]
@StephanMeijer @guyonroche I found that after setting both fgColor and bgColor to the same value I get my intended fill color with black text.  

`currentSheet.getRow(2).getCell(2).value = 'Test Text';
currentSheet.getRow(2).getCell(2).fill = {
        type: 'pattern',
        pattern: 'solid',
        fgColor:{argb:'FFFFFF00'},
        bgColor:{argb:'FFFFFF00'}
    };`

Will yield a yellow background with black text of "Test Text"

**Reply by AK221997** [reply]
this didnt worked for me. It yielded black blackground color with white text

**Comment by grumd** [comment, marked as answer]
fgColor is the color of your filling. you only need to specify fgColor if you're using `solid`

this is how you fill with red color:
```
{
  type: 'pattern',
  pattern: 'solid',
  fgColor: { argb:'FFFF0000' }
};
```

**Reply by AK221997** [reply]
where I can get other patterns for different color similar to this format argb:'FFFF0000' ?

**Reply by grumd** [reply]
It's hexadecimal format for colors in ARGB order. Alpha, Red, Green, Blue.

00 alpha is transparent, FF is opaque.

FF is red, 00 is no red.

So FF 00 FF 00 (without spaces) is green. You can google how to get any color in hex format, but what's unusual here is the first two letters being for alpha. Usually it's the last two letters.

**Reply by gersonmlb** [reply]
But the green color in ARGB -> 0000FF00 -> 00 is alpha transparent and the other colors have no problems. The problem is that many platforms give the hex format in RGBA and not in ARGB.

**Comment by fpaupier** [comment]
I think this issue is legit as I stumbled on the proper usage of cell filling today again. I opened a PR to clarify the doc about it.
https://github.com/exceljs/exceljs/pull/1649
